azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsInstance.java

```diff
@@ -77,4 +135,24 @@ private void updateStatus(EntityStatus status) {
                 .map(EventHubManagementClient::getEventHubs)
                 .ifPresent(c -> doModify(() -> c.createOrUpdate(getResourceGroupName(), namespace.getName(), getName(), inner.withStatus(status)), Status.UPDATING));
     }
+
+    private String getOrCreateConnectionString(List<AccessRights> accessRights) {
+        final List<EventHubAuthorizationRule> connectionStrings = Optional.ofNullable(getRemote())
+                .map(eventHubInstance -> eventHubInstance.listAuthorizationRules().stream()
+                        .filter(rule -> new HashSet<>(rule.rights()).containsAll(accessRights))
+                        .collect(Collectors.toList()))
+                .orElse(new ArrayList<>());
+        final EventHubsManager manager = getParent().getParent().getRemote();
+        if (connectionStrings.size() > 0) {
+            return connectionStrings.get(0).getKeys().primaryConnectionString();
+        }
+        if (Objects.isNull(manager)) {
```

Comment:
make `getOrCreateConnectionString` Nullable or throw an exception if manager is null?